xmppserver/src/main/java/org/jivesoftware/openfire/crowd/CrowdVCardProvider.java

```diff
@@ -74,12 +70,11 @@ public Element loadVCard(String username) {
                             .replace("@firstname@", user.firstName)
                             .replace("@email@", user.email)
                             .replace("@nickname@", username);
-                    vcard = readDocument(new StringReader(str)).getRootElement();
-                    
+                    vcard = SAXReaderUtil.readRootElement(str);
                 } catch (UserNotFoundException unfe) {
                     LOG.error("Unable to find user:" + String.valueOf(username) + " for loading its vcard", unfe);
                     return null;
-                } catch (DocumentException | SAXException e) {
+                } catch (ExecutionException | InterruptedException e) {
```

Comment:
Not sure how you're getting an InterruptedException here when you weren't before, but canonical handling of it requires you to `Thread.currentThread().interrupt()` - e.g. https://stackoverflow.com/a/3976377/359394

Reply:
Today I learned. Thanks!

Reply:
Like so? (change pushed)

Reply:
👍